app/controllers/events_controller.rb

```diff
@@ -0,0 +1,50 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user, only: %i[attend new]
+
+  def index
+    @past_events = Event.previous_events
+    @upcoming_events = Event.upcoming_events
+  end
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    user = User.find(session[:current_user_id])
+    @event = user.created_events.build(event_params)
+    pp @event
+    if @event.save
+      redirect_to event_path(@event), notice: 'Event Created successfully!'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
+
+  def attend
+    event = Event.find(params[:id])
+    event_to_attend = UsersEvent.where('user_id = ? AND event_id = ?', current_user.id, event.id)
+    if event_to_attend.count <= 0
+      a = UsersEvent.new
+      a.user_id = current_user.id
+      a.event_id = event.id
```

Comment:
Improve the code using the `build` method instead of assigning the columns' values manually, 

Hint: use the association `attended_events` of the `current_user` the same way you do with the event as and the user as a creator and passing the event id as a parameter.